### ISSUE TITLE
indicators = FALSE behavior

### ISSUE THREAD

**Issue by DavisVaughan** [DavisVaughan/hardhat#37]
These should all throw warnings of some kind.

Maybe when checking the formula RHS with `indicators = FALSE`, we should look for _only_ `+` and names, and warn about anything else! (rather than special casing everything)

``` r
library(hardhat)
library(gapminder)
gapminder <- gapminder[1:5,]
mold(year ~ year*continent, gapminder, indicators = FALSE)
#> $predictors
#> # A tibble: 5 x 2
#>    year continent
#>   <int> <fct>    
#> 1  1952 Asia     
#> 2  1957 Asia     
#> 3  1962 Asia     
#> 4  1967 Asia     
#> 5  1972 Asia     
#> 
#> $outcomes
#> # A tibble: 5 x 1
#>    year
#>   <int>
#> 1  1952
#> 2  1957
#> 3  1962
#> 4  1967
#> 5  1972
#> 
#> $preprocessor
#> Formula Preprocessor: 
#>  
#> # Predictors: 2 
#>   # Outcomes: 1 
#>    Intercept: FALSE 
#>   Indicators: FALSE 
#> 
#> $offset
#> NULL
mold(year ~ year*continent - year, gapminder, indicators = FALSE)
#> $predictors
#> # A tibble: 5 x 2
#>    year continent
#>   <int> <fct>    
#> 1  1952 Asia     
#> 2  1957 Asia     
#> 3  1962 Asia     
#> 4  1967 Asia     
#> 5  1972 Asia     
#> 
#> $outcomes
#> # A tibble: 5 x 1
#>    year
#>   <int>
#> 1  1952
#> 2  1957
#> 3  1962
#> 4  1967
#> 5  1972
#> 
#> $preprocessor
#> Formula Preprocessor: 
#>  
#> # Predictors: 2 
#>   # Outcomes: 1 
#>    Intercept: FALSE 
#>   Indicators: FALSE 
#> 
#> $offset
#> NULL
mold(year ~ (year+continent+pop)^2, gapminder, indicators = FALSE)
#> $predictors
#> # A tibble: 5 x 3
#>    year continent      pop
#>   <int> <fct>        <int>
#> 1  1952 Asia       8425333
#> 2  1957 Asia       9240934
#> 3  1962 Asia      10267083
#> 4  1967 Asia      11537966
#> 5  1972 Asia      13079460
#> 
#> $outcomes
#> # A tibble: 5 x 1
#>    year
#>   <int>
#> 1  1952
#> 2  1957
#> 3  1962
#> 4  1967
#> 5  1972
#> 
#> $preprocessor
#> Formula Preprocessor: 
#>  
#> # Predictors: 3 
#>   # Outcomes: 1 
#>    Intercept: FALSE 
#>   Indicators: FALSE 
#> 
#> $offset
#> NULL
```

<sup>Created on 2019-02-16 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1.9000)</sup>

**Comment by DavisVaughan** [comment]
Maybe not because we still want to do things like log(var)

**Comment by DavisVaughan** [comment]
Hard because you might want to allow `I(x * y)` to be computed algebraically and not represent interactions

**Comment by DavisVaughan** [comment]
Adding some notes on what led me to the right answer:

- when `indicators = FALSE`
- check that factor columns are only by themselves (factorcol + x) not factorcol:x or paste0(factorcol)
   - uses terms() directly on the formula, along with original factor classes to determine this
   - throws error if factor columns are used not by themselves
- remove the factor columns from the data
- remove the factor columns from the formula? (append `- factorcol`)
- run model.matrix(formula_no_factors, data_no_factors)
- matrix -> tibble
- add the factor columns back onto the data
   - Original positions are not kept, but this seems ok

**Comment by DavisVaughan** [comment]
Closed by #42
